src/web/services/course.service.ts

```diff
@@ -25,6 +25,38 @@ export class CourseService {
     return this.httpRequestService.get('/course', paramMap);
   }
 
+  /**
+   * Get student courses data of given google id in masquerade mode by calling API.
+   */
+  getStudentCoursesInMasqueradeMode(googleId: string): Observable<Courses> {
+    const paramMap: { [key: string]: string } = {
+      entitytype: 'student',
+      user: googleId,
+    };
+    return this.httpRequestService.get('/courses', paramMap);
+  }
+
+  /**
+   * Get instructor courses data of given google id in masquerade mode by calling API.
+   */
+  getInstructorCoursesInMasqueradeMode(googleId: string): Observable<Courses> {
+    const activeCoursesParamMap: { [key: string]: string } = {
+      coursestatus: 'active',
+      entitytype: 'instructor',
+      user: googleId,
+    };
+    const archivedCoursesParamMap: { [key: string]: string } = {
+      coursestatus: 'archived',
+      entitytype: 'instructor',
+      user: googleId,
+    };
+
+    return merge(
```

Comment:
The way of merge is incorrect.

This will give you an observable that returning two `Courses`.

Reply:
Try:
```javascript
forkJoin(
    this.httpRequestService.get('/courses', activeCoursesParamMap),
    this.httpRequestService.get('/courses', archivedCoursesParamMap),
).pipe(
    map((vals: Courses[]) => {
        return {
            courses: vals[0].courses.concat(vals[1].courses)
        }
    })
);
```